### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by elenapetra**
endpoints:
  GET - https://zq3zdypgs7.execute-api.eu-west-1.amazonaws.com/dev/products
  GET - https://zq3zdypgs7.execute-api.eu-west-1.amazonaws.com/dev/products/{productId}

### CODE REVIEW

File: src/mocks/data.ts [hunk 1]
```diff
@@ -3,47 +3,74 @@ import { CartItem } from "~/models/CartItem";
...
+//   },
+//   {
+//     description: "Short Product Description7",
+//     id: "7567ec4b-b10c-45c5-9345-fc73c48a80a1",
+//     price: 15,
+//     title: "Product 6",
+//   },
+// ];
+
+async function fetchProducts() {
```
Comment (dav-dio) [inline src/mocks/data.ts]: You don't need to add a fetchProducts method here. The code is pre-written in a way that you only have to add your backend (zq3zdypgs7) to apiPaths.ts. products.ts should then pick up on the path and initiate an Axios request.

Note that you may need to change the /product path to /products in the relevant products.ts methods. I recommend checking the console to see which endpoints are called.

File: src/mocks/data.ts [hunk 2]
```diff
@@ -3,47 +3,74 @@ import { CartItem } from "~/models/CartItem";
...
+    return [];
+  }
+}
+
+export const products = await fetchProducts();
+console.log(products);
 
 export const availableProducts: AvailableProduct[] = products.map(
-  (product, index) => ({ ...product, count: index + 1 })
+  (product: any, index: number) => ({ ...product, count: product.count })
```
Comment (dav-dio) [inline src/mocks/data.ts]: I don't think this is necessary. This method is part of the Mock Service Worker implementation that is supposed to fill the page with mock data. A working request (via products.ts - useAvailableProducts()) should disable the MSW for that endpoint.

**Review by dav-dio — CHANGES_REQUESTED** [review]

File: src/mocks/data.ts (hunk 2 above)
Comment (elenapetra) [reply src/mocks/data.ts]: I've changed the code to its original state. Thanks for the explanation, it makes sense now.

File: src/mocks/data.ts (hunk 1 above)
Comment (elenapetra) [reply src/mocks/data.ts]: I've made the necessary changes in products.ts and apiPaths.ts files to fetch data from corresponding endpoints.

**Review by dav-dio — APPROVED** [review]

**Pull request merged**